javaagent-extension-api/src/main/java/io/opentelemetry/javaagent/extension/AgentListener.java

```diff
@@ -22,25 +21,26 @@
 public interface AgentListener extends Ordered {
 
   /**
-   * Runs before {@link AgentBuilder} construction, before any instrumentation is added. Not called
-   * if noop api enabled via {@code otel.javaagent.experimental.use-noop-api}.
-   *
-   * <p>Execute only minimal code because any classes loaded before the agent installation will have
-   * to be retransformed, which takes extra time, and more importantly means that fields can't be
-   * added to those classes - which causes {@link VirtualField} to fall back to the less performant
-   * cache-based implementation for those classes.
+   * Runs after instrumentations are added to {@link AgentBuilder} and after the agent is installed
+   * on an {@link Instrumentation}. Not called if noop api enabled via {@code
+   * otel.javaagent.experimental.use-noop-api}.
    *
-   * @deprecated This method will be removed in the next release.
+   * @deprecated Implement {{@link #afterAgent(AutoConfiguredOpenTelemetrySdk)}} instead.
    */
   @Deprecated
-  default void beforeAgent(
-      Config config, AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {}
+  default void afterAgent(
+      Config config, AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
+    throw new UnsupportedOperationException(
+        "This method is deprecated and will be removed in a future release;"
+            + " implement AgentListener#afterAgent(AutoConfiguredOpenTelemetrySdk) instead");
+  }
 
   /**
    * Runs after instrumentations are added to {@link AgentBuilder} and after the agent is installed
    * on an {@link Instrumentation}. Not called if noop api enabled via {@code
    * otel.javaagent.experimental.use-noop-api}.
    */
-  default void afterAgent(
-      Config config, AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {}
+  default void afterAgent(AutoConfiguredOpenTelemetrySdk autoConfiguredOpenTelemetrySdk) {
+    afterAgent(Config.get(), autoConfiguredOpenTelemetrySdk);
+  }
```

Comment:
> are you thinking to drop the default after the next release?

Yes, that was my intention 👍 

> and do you think we're confident after that to mark it with `@FunctionInterface` or better not to?

Hm, it's an SPI, it can't be a lambda function -- probably better not to add it.

Reply:
> it's an SPI, it can't be a lambda function

🤦‍♂️ 👍